aten/src/ATen/native/hammerblade/Addmm.cpp

```diff
@@ -12,13 +13,24 @@ Tensor addmm_hb(
     Scalar alpha
 ) {
 
-  TORCH_CHECK(false, "addmm_hb not implemented");
-  //TODO: implmement the correct addmm_hb host code
-  //      you may use void offload_tensor_scalar_impl(std::vector<Tensor>, std::vector<Scalar>,
-  //                                                  cosn char*)
-  //      which is defined in Offload.h for kernel offloading
-  return self;
+  if ( (self.scalar_type() != ScalarType::Float)
+    || (mat1.scalar_type() != ScalarType::Float)
+    || (mat2.scalar_type() != ScalarType::Float) ) {
+    AT_ERROR("HammerBlade addmm is implemented for Float only");
+  }
 
+  using scalar_t = float;
+
+  TORCH_CHECK(mat1.size(1) == mat2.size(0), "Argument #3: Expected dim 0 size ", mat1.size(1), ", got ", mat2.size(0));
```

Comment:
This is a nice reference for argument checks required for addmm, I think a couple more checks also makes sense:
https://github.com/cornell-brg/hb-pytorch/blob/d5358b65dd034b15d5a8d593436966c11b0361be/aten/src/TH/generic/THTensorMath.cpp#L130-L147

Reply:
Actually the `expand_size` function takes care of broadcasting/mismatched kernels here:

Reply:
I think it might be still useful to check if mat1 and mat2 are 2D tensors (matrices)?

Reply:
Added!